sklearn/ensemble/bagging.py

```diff
@@ -387,6 +397,56 @@ def _validate_y(self, y):
         # Default implementation
         return column_or_1d(y, warn=True)
 
+    def _get_estimators_data_draws(self, sample_mask=False, feature_mask=False):
+        # If max_samples is float
+        if not isinstance(self.max_samples, (numbers.Integral, np.integer)):
```

Comment:
why do we need to do input validation here again? This is an internal method? I think we should store ``max_sample`` and ``max_features`` in ``self._max_samples`` and ``self._max_features`` during fit and use these here.